pallets/funding/src/tests/2_evaluation.rs

```diff
@@ -0,0 +1,552 @@
+
+			inst.execute(|| {
+				assert_noop!(
+					PolimecFunding::start_evaluation(
+						RuntimeOrigin::signed(issuer + 1),
+						get_mock_jwt(issuer + 1, InvestorType::Institutional, generate_did_from_account(issuer + 1)),
+						project_id
+					),
+					Error::<TestRuntime>::NotAllowed
+				);
+			});
+		}
+	}
+}
+
+#[cfg(test)]
+mod evaluate_extrinsic {
+	use super::*;
+
+	#[cfg(test)]
+	mod success {
+		use super::*;
+
+		#[test]
+		fn all_investor_types() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let issuer = ISSUER_1;
+			let project_metadata = default_project_metadata(inst.get_new_nonce(), issuer);
+			let project_id = inst.create_evaluating_project(project_metadata.clone(), issuer);
+
+			let evaluations = vec![
+				(EVALUATOR_1, 500 * US_DOLLAR).into(),
+				(EVALUATOR_2, 1000 * US_DOLLAR).into(),
+				(EVALUATOR_3, 20_000 * US_DOLLAR).into(),
+			];
+			let necessary_plmc = MockInstantiator::calculate_evaluation_plmc_spent(evaluations.clone());
+			let plmc_existential_deposits = necessary_plmc.accounts().existential_deposits();
+
+			inst.mint_plmc_to(necessary_plmc);
+			inst.mint_plmc_to(plmc_existential_deposits);
+
+			assert_ok!(inst.execute(|| PolimecFunding::evaluate(
+				RuntimeOrigin::signed(evaluations[0].account),
+				get_mock_jwt(
+					evaluations[0].account,
+					InvestorType::Institutional,
+					generate_did_from_account(evaluations[0].account)
+				),
+				project_id,
+				evaluations[0].usd_amount,
+			)));
+
+			assert_ok!(inst.execute(|| PolimecFunding::evaluate(
+				RuntimeOrigin::signed(evaluations[1].account),
+				get_mock_jwt(
+					evaluations[1].account,
+					InvestorType::Professional,
+					generate_did_from_account(evaluations[1].account)
+				),
+				project_id,
+				evaluations[1].usd_amount,
+			)));
+
+			assert_ok!(inst.execute(|| PolimecFunding::evaluate(
+				RuntimeOrigin::signed(evaluations[2].account),
+				get_mock_jwt(
+					evaluations[2].account,
+					InvestorType::Retail,
+					generate_did_from_account(evaluations[2].account)
+				),
+				project_id,
+				evaluations[2].usd_amount,
+			)));
+		}
+
+		#[test]
+		fn using_frozen_tokens() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let issuer = ISSUER_1;
+			let project_metadata = default_project_metadata(inst.get_new_nonce(), issuer);
+			let project_id = inst.create_evaluating_project(project_metadata.clone(), issuer);
+
+			let evaluation = UserToUSDBalance::new(EVALUATOR_1, 500 * US_DOLLAR);
+			let necessary_plmc = MockInstantiator::calculate_evaluation_plmc_spent(vec![evaluation.clone()]);
+			let plmc_existential_deposits = necessary_plmc.accounts().existential_deposits();
+
+			inst.mint_plmc_to(necessary_plmc.clone());
+			inst.mint_plmc_to(plmc_existential_deposits);
+
+			inst.execute(|| {
+				frame_system::Account::<TestRuntime>::mutate(EVALUATOR_1, |account| {
+					account.data.frozen = necessary_plmc[0].plmc_amount;
+				});
+			});
```

Comment:
Would prefer to see a freeze using NativeCurrency (or via the implemented pallet directly) so we are actually checking against the correct interface. All other pallets use this pallet balances interface as well, so we do not know if something else would prevent us from using frozen tokens in pallet balances.

Reply:
Was inaccessible through the currency type on funding pallet, but I guess since I have access to the mock runtime I can do it from there